### PR TITLE
Save for later: no results placeholder

### PR DESCRIPTION
**Pull request by ctarda**
Implements #9411 

Present a placeholder when there are no posts saved for later. 

I didn't add an accessory view because we can save posts for later from at least three different lists (Followed sites, Search and My Likes).

simulator screen shot - iphone 8 - 2018-05-20 at 18 38 01

To test:
0. Activate the feature flag
1. Navigate to the Saved Posts list
2. Remove previously saved posts (if any) until the list is empty

### CODE REVIEW

**Comment by frosty** [comment]
@iamthomasbishop Is this good with you, or would you like any tweaks?

**Comment by iamthomasbishop** [comment]
Just a quick copy recommendations, although we might want to have that copy-reviewed to confirm it makes sense 😄:

* Can we change the title from `No posts saved for later yet` to `No posts saved yet`

* Can we change the subtitle to `Tap the [gridicons-bookmark] icon on a post to save it.` 

I'd also like to find an illustration for this empty state, but I'm not seeing one that's immediately relevant and available, so we may have to roll a custom illustration for it. If waiting for a new illustration is a blocker, let's roll with this, and create a separate issue to add an appropriate illustration asap.

**Comment by frosty** [comment]
@iamthomasbishop Sounds good! How does this look?

screen shot 2018-05-23 at 11 37 56

**Comment by frosty** [comment]
Also, this feature won't be shipping for a little while yet so we have time to merge this now and add an illustration if we want one later. Are you able to wrangle an illustration? If we even need one?

File: WordPress/Classes/Extensions/NSMutableAttributedString+Helpers.swift [hunk 1]
```diff
@@ -22,4 +22,28 @@ extension NSMutableAttributedString {
...
+        guard range.location != NSNotFound else {
+            return
+        }
+
+        let font = attribute(NSAttributedStringKey.font, at: range.location, effectiveRange: nil) as? UIFont
+        let capHeight = font?.capHeight ?? 0
+
+        let attachment = NSTextAttachment()
+        attachment.image = icon
+        attachment.bounds = CGRect(x: 0.0, y: (capHeight - icon.size.height).rounded() / 2, width: icon.size.width, height: icon.size.height)
```
Comment (jleandroperez) [inline WordPress/Classes/Extensions/NSMutableAttributedString+Helpers.swift]:
```
y: (capHeight - icon.size.height).rounded() / 2
```

We *probably*  want to actually apply the round after the division

**Comment by iamthomasbishop** [comment]
@frosty I do like having the icon there, wondering if it should be slightly smaller to sit nicely inline. Is the icon currently at 24pt? If so, let's try 18pt.

Now that I see it, I also think the copy changes I requested read a little awkward 😬 . Does `Tap [gridicons-bookmark-outline] to save a post to your list.` sound better? // @etoledom - will this cause issues w/ accessibility? If it's okay, let's go with that and then request a copy review.

If the above would cause issues, we should divert to something like `Tap on the bookmark icon ( [gridicons-bookmark-outline] ) to save a post to your list.`, then request copy review 😄

**Comment by kristastevens** [comment]
Hi @frosty -- thanks so much for the excellent context around this text -- it's so helpful! 

I think what you've got looks good. I have one suggestion: 

> No posts saved yet 

Could become: 

> No posts saved -- yet! 

I feel like it's just a tiny bit friendlier, and the em dash serves as a subtle call to action to get the user to save a post. Either way, I do feel like that sentence needs terminal punctuation, so if you choose to go with the original, it needs a period at the end. 

Of the two options for the "tap" line, I like the second best -- it's clean and clear: 

> "Tap [bookmark icon] to save post to your list."

Thanks so much for asking us to weigh in!

**Comment by frosty** [comment]
Thanks @kristastevens! Sounds good to me!

**Comment by frosty** [comment]
@jleandroperez Ready for one more quick review if you don't mind! I'll tackle the accessibility of the view in a separate PR, as I need to go over the accessibility of Save for Later as a whole anyway.

**Pull request merged**